sdk/nodejs/runtime/closure/parseFunction.ts

```diff
@@ -140,66 +140,27 @@ function parseFunctionCode(funcString: string): [string, ParsedFunctionCode] {
+    // There are three general forms of node toString'ed Functions we're trying to find out here.
     //
-    // First we check if we have a open curly or not.  If we don't, then we're in the last case. We
-    // confirm that we have a => (throwing if we don't).
+    // 1. `[mods] (...) => ...
     //
-    // If we do have an open curly, then we're in one of the top two cases.  To determine which we
-    // trim things up to the open curly, leaving us with either:
+    //      i.e. an arrow function.  We need to ensure that arrow-functions stay arrow-functions,
+    //      and non-arrow-functions end up looking like normal `function` functions. This will make
+    //      it so that we can correctly handle 'this' properly depending on if that should be
+    //      treated as the lexical capture of 'this' or the non-lexical 'this'.
     //
-    //      (...) {
-    //      (...) => {
-    //
-    // We then see if we have an => or not.  if we do, it's a lambda.  If we don't, it's a function
-    // with a computed name.
-
-    // Note: Using indexOf is problematic here as we can be thrown off by random code in the func
-    // string (for example comments).  When using indexOf, a justification should be given as to
-    // why this approach is more often correct than not.
-
-    // There are three general forms of node toString'ed Functions we're trying to find out here.
-    //
-    // 1. `class Foo { ... }`
+    // 2. `class Foo { ... }`
     //
     //      i.e. node uses the entire string of a class when toString'ing the constructor function
     //      for it.
     //
-    // 2. `[mods] function ...
+    // 3. `[mods] function ...
     //
     //      i.e. a normal function (maybe async, maybe a get/set function, but def not an arrow
     //      function)
-    //
-    // 3. `[mods] (...) => ...
-    //
-    //      i.e. an arrow function.  Harder to figure out as we have to look further ahead (weakly)
-    //      for the `=>` portion.
-
 
-    // Using indexOf("{") is mostly acceptable here.  This is purely to see if we're a class or
-    // a function with a body.  If we do *not* find a curly *at all*, then this really could
-    // only be a lambda, and that's what this block is trying to check for.  If we do find an
-    // open curly, we attempt to figure out what sort of entity we are below.
-    const openCurlyIndex = funcString.indexOf("{");
-    if (openCurlyIndex < 0) {
-        // No block body.  Can happen if this is an arrow function with an expression body.
-        const arrowIndex = funcString.indexOf("=>");
-        if (arrowIndex >= 0) {
-            // (...) => expr
-            return ["", { funcExprWithoutName: funcString, isArrowFunction: true }];
-        }
-
-        return [`the function form was not understood.`, <any>undefined];
+    if (tryParseAsArrowFunction(funcString)) {
+        return ["", { funcExprWithoutName: funcString, isArrowFunction: true }];
     }
```

Comment:
this is the bulk of hte change.  instead of just doing string checks, we literally just let the TS parser try to handle arrow functions.  This just sidesteps all the problems with our lookahead code and how we tried to determine what was going on depending on if `=>` or `{` was present.

Reply:
this actually just makes things much simpler.  We now do the same things for arrow functions and classes.  And, in the future, we'll likely move to this approach for all toString'ed functions.